### PR TITLE
bugfix/minimum pegs should be less than or equal to limit

### LINKED ISSUE #99: Minimum can be set larger than peg limit
**Describe the bug**
It is possible to use the `numberconfig` to set the minimum number of pegs required to win to be *higher* than the maximum number of pegs able to be given out in a given cycle.

**To Reproduce**
1. `@PockyBot numberconfig set limit 10`
2. `@PockyBot numberconfig set minimum 11`
3. Observe that no error occurs

**Expected behavior**
The minimimum number of pegs required to win should be unable to be set higher than the current peg limit. An error should be reported when attempting to set it higher.

**Screenshots**
image

### PR DESCRIPTION
**Pull request by gillianmilne**
Closes #99

### CODE REVIEW

File: src/lib/response-triggers/numberconfig.ts [hunk 1]
```diff
@@ -106,4 +104,21 @@ export default class NumberConfig extends Trigger {
 
 		return message;
 	}
+
+	private validateConfigActionSet(words: string[]) : string {
```
Comment (Evangelinexx) [inline src/lib/response-triggers/numberconfig.ts]: +1 to refactoring this into a validate method

File: src/lib/response-triggers/numberconfig.ts [hunk 2]
```diff
@@ -106,4 +104,21 @@ export default class NumberConfig extends Trigger {
...
+		}
+
+		const value = Number(words[3]);
+
+		if (isNaN(value)) {
+			return 'Config must be set to a number';
+		}
+
+		if(value < 0) {
+			return 'Config should be greater than 0.';
```
Comment (jimcullenaus) [inline src/lib/response-triggers/numberconfig.ts]: "Should be no less than 0" or "should be greater than or equal to 0", surely?

(Also, you've missed the space after `if` repeatedly in this file.)

File: src/spec/numberconfig-spec.ts [hunk 3]
```diff
@@ -122,6 +124,20 @@ test | 1
...
+		const configMinimumMessage = { text: `${constants.botName} numberconfig set minimum 6` };
+		let minResponse = await configuration.createMessage(configMinimumMessage);
+		expect(minResponse.markdown).toBe('Minimum pegs must be less than or equal to maximum pegs.');
+		done();
+	});
+
+	it('should fail to set minimum less than 0', async (done : DoneFn) => {
+		const configMinimumMessage = { text: `${constants.botName} numberconfig set minimum -1` };
+		let minResponse = await configuration.createMessage(configMinimumMessage);
+		expect(minResponse.markdown).toBe('Config should be greater than 0.');
```
Comment (jimcullenaus) [inline src/spec/numberconfig-spec.ts]: If the error message changes above, the test will need to be changed to match.

File: src/lib/response-triggers/numberconfig.ts (hunk 2 above)
Comment (gillianmilne) [reply src/lib/response-triggers/numberconfig.ts]: buh, I thought I had fixed the wording. Guess I didn't

File: src/spec/numberconfig-spec.ts [hunk 4]
```diff
@@ -38,7 +38,9 @@ beforeAll(() => {
 		if (config === 'test') {
 			return 1;
 		}
-
+		if (config === 'limit') {
```
Comment (drewfreyling) [inline src/spec/numberconfig-spec.ts]: Couldn't this be combined with the previous if statement?

File: package.json [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {
 	"name": "PockyBot",
-	"version": "1.7.2",
-	"description": "Webex bot that handles team recognition",
+	"version": "1.7.3",
+	"description": "Spark bot that handles team recognition",
```
Comment (Lauraducky) [inline package.json]: Changing this back?

**Pull request merged**